R/composeVar.R

```diff
@@ -33,17 +34,29 @@
 #'
 #'
 #'@export
-composeVar <- function(GADSdat, sourceVars, primarySourceVar, newVar) {
+composeVar <- function(GADSdat, sourceVars, primarySourceVar, newVar, checkVarName = TRUE) {
   UseMethod("composeVar")
 }
 #'@export
-composeVar.GADSdat <- function(GADSdat, sourceVars, primarySourceVar, newVar) {
+composeVar.GADSdat <- function(GADSdat, sourceVars, primarySourceVar, newVar, checkVarName = TRUE) {
   check_GADSdat(GADSdat)
   check_vars_in_GADSdat(GADSdat, vars = sourceVars)
-  if(!is.character(sourceVars) || length(sourceVars) != 2) stop("'sourceVars' must be a character vector of length 2.")
-  if(!is.character(primarySourceVar) || length(primarySourceVar) != 1) stop("'primarySourceVar' must be a character vector of length 1.")
-  if(!is.character(primarySourceVar) || length(newVar) != 1) stop("'newVar' must be a character vector of length 1.")
-  if(!primarySourceVar %in% sourceVars) stop("'primarySourceVar' must be a name in 'sourceVars'.")
+  check_logicalArgument(checkVarName, argName = "checkVarName")
+  if(!is.character(sourceVars) || length(sourceVars) != 2) {
+    stop("'sourceVars' must be a character vector of length 2.")
+  }
+  if(!is.character(primarySourceVar) || length(primarySourceVar) != 1) {
+    stop("'primarySourceVar' must be a character vector of length 1.")
+  }
+  if(!is.character(primarySourceVar) || length(newVar) != 1) {
+    stop("'newVar' must be a character vector of length 1.")
+  }

```

Comment:
I agree with all comments and have substituted the manual check with the existing `check_characterArgument()`. A more elaborate `check_argument()` function as suggested seems like a good idea and useful for all (?) `eat`-packages. My proposition would be to add this to eatTools (see weirichs/eatTools#6).

Reply:
I'm not sure about other `eat`-packages, but for `eatGADS ` there are many use cases as far as I remember (maybe, it's biased because I've particularly looked for it).

Reply:
`eatPrepTBA` might be a good outlet. In `eatPrep`, there were not so many type checks (in the few functions I have worked on) as far as I recall (but it's been a while since I last worked on `eatPrep`) 🤔 But I think, these would be useful additions and I see a lot of poential use cases. These `check_arguments()` would be explicit, and informative input checks that can readily be used. So, potentially: Yes!